### PR TITLE
[i18n] i18n improvements

### PR DESCRIPTION
**Pull request by ramiy**
1. Add translators comments to explain %s / %d placeholders.
2. Replace wrong use of _x() with translators comments. The _x() used for context, not examples, instead you should use translators comments.

### CODE REVIEW

**Comment by ramiy** [comment]
Old string with placeholder:
`__( 'Add Ons for %s', 'freemius' ),`

New:
`/* translators: %s: Plugin name */`
`__( 'Add Ons for %s', 'freemius' ),`

----

Old string with _x():
`_x( '%s Plan', 'e.g. Professional Plan', 'freemius' ),`

New:
`/* translators: %s: Plan name (e.g. Professional) */`
`__( '%s Plan', 'freemius' ),`

**Comment by vovafeldman** [comment]
Hey @ramiy - thanks for reviewing the translations. Can you please explain the 2nd type of changes:

Old string with _x():
`_x( '%s Plan', 'e.g. Professional Plan', 'freemius' ),`

New:
`/* translators: %s: Plan name (e.g. Professional) */`
`__( '%s Plan', 'freemius' ),`

Based on the PHP Doc description of `_x()` it says that `$context` is *Context information for the translators.*. What am I missing?

**Comment by ramiy** [comment]
@vovafeldman 

Generally, the **context** is used to describe the entire string, **translators comment** are used to describe specific placeholders like `%s`, `%1$s`, `%2$s` and others.



----

Also, in WordPress core, we use the context function when we have two identical strings with different meaning.

The context allows us to create two separate entries in [translate.wordpress.org](https://translate.wordpress.org/).

----

For example:

First string:
`/* translators: 1: user_login, 2: display name */`
`_x( '%1$s, %2$s', 'user' )`

Second string:
`/* translators: 1: post title, 2: post date */`
`_x( '%1$s, %2$s', 'post' )`

When using the `_x()` functions, we will see two separate entries in [translate.wordpress.org](https://translate.wordpress.org/), to allow translators to enter different translations for each context. It's useful for languages where they need to show the `%2$s` before `%1$s` or replace the comma with colon, but only in the first string.

If we use `__()` for both strings, translators wont be able to provide different translations for the two strings. With `_x()` they can.

**Comment by vovafeldman** [comment]
Thanks @ramiy! You made a lot of suggestions so we'll try to review it next week.

File: includes/i18n.php [hunk 1]
```diff
@@ -245,96 +278,135 @@
...
-		'latest-x-version'                         => _x( 'the latest %s version here', '%s - plan name, as the latest professional version here', 'freemius' ),
-		'trial-x-promotion-message'                => __( 'How do you like %s so far? Test all our %s premium features with a %d-day free trial.', 'freemius' ),
+		'trial-cancel-message'                     => /* translators: %s: Plugin name */
+		                                              __( 'Your %s free trial was successfully cancelled.', 'freemius' ),
+		'version-x-released'                       => /* translators: %s: Numeric version number */
+		                                              __( 'Version %s was released.', 'freemius' ),
+		'please-download-x'                        => /* translators: %s: Plugin name */
+		                                              __( 'Please download %s.', 'freemius' ),
+		'latest-x-version'                         => /* translators: %s: Plan name (e.g. professional) */
+		                                              _x( 'the latest %s version here', 'freemius' ),
```
Comment (vovafeldman) [inline includes/i18n.php]: @ramiy I believe the `_x` should be `__`.

File: includes/i18n.php [hunk 2]
```diff
@@ -175,11 +201,16 @@
...
-		'thanks-x'                                 => _x( 'Thanks %s!', 'a greeting. E.g. Thanks John!', 'freemius' ),
-		'connect-message'                          => __( 'In order to enjoy all our features and functionality, %s needs to connect your user, %s at %s, to %s', 'freemius' ),
-		'connect-message_on-update'                => __( 'Please help us improve %2$s! If you opt-in, some data about your usage of %2$s will be sent to %5$s. If you skip this, that\'s okay! %2$s will still work just fine.', 'freemius' ),
-		'pending-activation-message'               => __( 'You should receive an activation email for %s to your mailbox at %s. Please make sure you click the activation button in that email to complete the install.', 'freemius' ),
+		'hey-x'                                    => /* translators: %s: User name (e.g. John) */
+		                                              __( 'Hey %s,', 'freemius' ),
+		'thanks-x'                                 => /* translators: %s: User name (e.g. John) */
+		                                              __( 'Thanks %s!', 'freemius' ),
+		'connect-message'                          => /* translators: 1: Plugin name 2: ??? 3: ??? 4: ??? */
```
Comment (vovafeldman) [inline includes/i18n.php]: @ramiy:

2: Admin username
3: Current site's domain
4: Link to `freemius.com`

File: includes/i18n.php [hunk 3]
```diff
@@ -70,13 +76,20 @@
...
+		                                              __( 'Up to %s Sites', 'freemius' ),
+		'renew-license-now'                        => /* translators: 1: Starting HTML tag 2: Starting HTML tag 3: Plan name */
+		                                              __( '%sRenew your license now%s to access version %s features and support.', 'freemius' ),
+		'x-plan'                                   => /* translators: %s: Plan name (e.g. professional) */
+		                                              __( '%s Plan', 'freemius' ),
+		'you-are-step-away'                        => /* translators: %s: Plan name (e.g. professional) */
+		                                              __( 'You are just one step away - %s', 'freemius' ),
+		'activate-x-now'                           => /* translators: %s: Plugin name */
+		                                              __( 'Complete "%s" Activation Now', 'freemius' ),
+		'few-plugin-tweaks'                        => /* translators: %s: Plugin name */
```
Comment (vovafeldman) [inline includes/i18n.php]: @ramiy `%s: additional suffix string`

File: includes/i18n.php [hunk 4]
```diff
@@ -70,13 +76,20 @@
...
-		'license-x-sites'                          => __( 'Up to %s Sites', 'freemius' ),
-		'renew-license-now'                        => __( '%sRenew your license now%s to access version %s features and support.', 'freemius' ),
-		'x-plan'                                   => _x( '%s Plan', 'e.g. Professional Plan', 'freemius' ),
-		'you-are-step-away'                        => __( 'You are just one step away - %s', 'freemius' ),
-		'activate-x-now'                           => _x( 'Complete "%s" Activation Now', '%s - plugin name. As complete "Jetpack" activation now', 'freemius' ),
-		'few-plugin-tweaks'                        => __( 'We made a few tweaks to the plugin, %s', 'freemius' ),
-		'optin-x-now'                              => __( 'Opt-in to make "%s" Better!', 'freemius' ),
+		'license-x-sites'                          => /* translators: %s: Number of licenses */
+		                                              __( 'Up to %s Sites', 'freemius' ),
+		'renew-license-now'                        => /* translators: 1: Starting HTML tag 2: Starting HTML tag 3: Plan name */
```
Comment (vovafeldman) [inline includes/i18n.php]: @ramiy `2: Ending HTML tag`

File: includes/i18n.php [hunk 5]
```diff
@@ -47,20 +47,26 @@
...
 		'free-trial'                               => __( 'Free Trial', 'freemius' ),
-		'start-free-x'                             => __( 'Start my free %s', 'freemius' ),
-		'no-commitment-x'                          => __( 'No commitment for %s - cancel anytime', 'freemius' ),
-		'after-x-pay-as-little-y'                  => __( 'After your free %s, pay as little as %s', 'freemius' ),
+		'start-free-x'                             => /* translators: %s: Trial name */
+		                                              __( 'Start my free %s', 'freemius' ),
+		'no-commitment-x'                          => /* translators: %s: Plan name (e.g. professional) */
+		                                              __( 'No commitment for %s - cancel anytime', 'freemius' ),
+		'after-x-pay-as-little-y'                  => /* translators: 1: Trial name 2: Price */
+		                                              __( 'After your free %s, pay as little as %s', 'freemius' ),
```
Comment (vovafeldman) [inline includes/i18n.php]: @ramiy  `1: Trial name (e.g. '7-day Pro trial')`

File: includes/i18n.php [hunk 6]
```diff
@@ -47,20 +47,26 @@
 		'downgrade'                                => _x( 'Downgrade', 'verb', 'freemius' ),
 		'cancel-trial'                             => __( 'Cancel Trial', 'freemius' ),
 		'free-trial'                               => __( 'Free Trial', 'freemius' ),
-		'start-free-x'                             => __( 'Start my free %s', 'freemius' ),
-		'no-commitment-x'                          => __( 'No commitment for %s - cancel anytime', 'freemius' ),
-		'after-x-pay-as-little-y'                  => __( 'After your free %s, pay as little as %s', 'freemius' ),
+		'start-free-x'                             => /* translators: %s: Trial name */
+		                                              __( 'Start my free %s', 'freemius' ),
+		'no-commitment-x'                          => /* translators: %s: Plan name (e.g. professional) */
```
Comment (vovafeldman) [inline includes/i18n.php]: @ramiy `%s: Trial length (e.g. '7 days')`

**Comment by vovafeldman** [comment]
@ramiy I've checked all the code suggestions and added some minor improvements. Please update, resubmit the PR and I'll merge.

**Comment by ramiy** [comment]
@vovafeldman updated.

**Pull request closed without merging**